### PR TITLE
Update docs to point to new docs on hexpm

### PR DESCRIPTION
**Pull request by starbelly**

### CODE REVIEW

**Review by ferd — COMMENTED** [review]
There may be external references pointing to these pages. It may make more sense to keep a quick intro of things and then refer to the external doc to at least avoid leaving people with no way to bridge the gap from a hyperlink they clicked onto into a 404 on the docsite.

**Comment by starbelly** [comment]
> There may be external references pointing to these pages. It may make more sense to keep a quick intro of things and then refer to the external doc to at least avoid leaving people with no way to bridge the gap from a hyperlink they clicked onto into a 404 on the docsite.

Excellent point, will adjust.

**Comment by starbelly** [comment]
@ferd  How about we lose the extra pages, redirect those pages to `_index.md`, and  keep a copy of the README.me (so to speak) in the `_index.md` ?

**Comment by ferd** [comment]
Redirects are probably fine. We should use non-permanent ones I guess.

**Comment by starbelly** [comment]
> Redirects are probably fine. We should use non-permanent ones I guess.

I've added aliases, these are not permanent redirects, rather `<link rel="canonical" href="/docs/package_management/"/>` and followed (by decent browsers). 

Now, what we can do is mix in some of the content from hex.pm, and I can add that to the rebar3_hex README as well, so anytime things get updated a copy/paste job can ensue essentially. 

I initially didn't want to double up on docs here and there, but perhaps it makes sense... thoughts?

File: content/en/docs/package_management/_index.md [hunk 1]
```diff
@@ -4,5 +4,170 @@ linkTitle: "Hex Package Management"
...
+    - /docs/package_management/hex_package_management/
+    - /docs/package_management/publishing-packages/
 ---
 
+
+For package management `rebar3` uses [hex.pm](https://hex.pm), a package manager for Erlang and Elixir packages.
+
+## Basic usage
+
+You should read the [docs](https://hexdocs.pm/rebar3_hex/) for a complete over for each provider, but below is a
```
Comment (ferd) [inline content/en/docs/package_management/_index.md:18]:
```suggestion
You should read the [docs](https://hexdocs.pm/rebar3_hex/) for a complete guide for each provider, but below is a
```

**Review by ferd — APPROVED** [review]

**Pull request merged**